src/containers/Sidebar/index.js

```diff
@@ -22,6 +27,49 @@ const Sidebar = (props) => {
 		return () => window.removeEventListener('resize', handleResize);
 	}, [windowDimensions]);
 
+	useEffect(() => {
```

Comment:
@Aringoaway Ok,everything from line 30 to line 71 can be simplified as:

```
useEffect(() => {
	if (windowDimensions.width && (windowDimensions.width > 768) && (windowDimensions.width <= 1024)) {
		dispatch({
			type: COLLAPSE_SIDEBAR,
			isSidebarCollapsed: true
		});
	} else {
		dispatch({
			type: COLLAPSE_SIDEBAR,
			isSidebarCollapsed: false
		});
	}
}, [windowDimensions.width])
```

Its that simple :) 

You are doing unnecessary dispatching based on state, which will have an impact on performance and thus it is not sustainable.

Reply:
@Pe5h4 I understand, I also wrote the code this way at first. But I was confused by the behaviour of the sidebar. If I open it and start reducing the screen size, it keeps closing. Is this the correct behaviour shown in the video below? I've been trying to achieve the other behaviour shown in the video in the description
https://user-images.githubusercontent.com/65866093/219315266-afb3498f-d62a-4a61-9cab-d8991c1c25f5.mp4

Reply:
@Aringoaway  Yes, this should be the correct behaviour :)